stdlib/marshal.mli

```diff
@@ -109,6 +109,8 @@ val to_channel : out_channel -> 'a -> extern_flags list -> unit
    when read back on a 32-bit platform.  The [Mashal.Compat_32] flag
    only matters when marshaling is performed on a 64-bit platform;
    it has no effect if marshaling is performed on a 32-bit platform.
+   
+   @raise Failure if [chan] is not in binary mode.
```

Comment:
Why not add the same info to `Marshal.from_channel`?  It also fails if the  channel is not in binary mode.

Reply:
Sure, I must have overlooked that.

Reply:
Would you please add the info as I suggested, instead of resolving this conversation and doing nothing?

Reply:
Sorry, my comment above was misleading. I already did that change in my branch, so it should be included in this PR (see commit [e6d11e0](https://github.com/ocaml/ocaml/pull/10012/commits/e6d11e02c4747a5e56581743051c613ed4251ca4) ).

Reply:
Sorry for the possible misunderstanding, but I just downloaded your branch, and in `stdlib/marshal.mli` I see:
```
   @raise Failure if [chan] is not in binary mode.
```
in the documentation comment for `Marshal.to_channel`, but I do not see it in the documentation comment for `Marshal.from_channel`, which I copy verbatim below for reference:
```
val from_channel : in_channel -> 'a
(** [Marshal.from_channel chan] reads from channel [chan] the
   byte representation of a structured value, as produced by
   one of the [Marshal.to_*] functions, and reconstructs and
   returns the corresponding value.

   @raise End_of_file if [chan] is already at the end of the file.

   @raise Failure if the end of the file is reached during
   unmarshalling itself. *)
```
So, would you please add `@raise Failure if [chan] is not in binary mode.` to the documentation comment for `Marshal.from_channel` as I already requested TWICE?  Thank you.

Reply:
Oops, now I got it. Sorry again, I was referring to the quoted line numbers instead of looking up the cited function. Should be _really_ added now. I did some rewording, as `Marshal.form_channel` might also throw `Failure` for another reason.